src/lbaf/IO/configValidator.py

```diff
@@ -0,0 +1,123 @@
+        self.__config_to_validate = config_to_validate
+        self.__skeleton = Schema({
+            Or("from_data", "from_samplers", only_one=True): dict,
+            "work_model": {"name": And(str, lambda c: c in ("LoadOnly", "AffineCombination"),
+                                       error="LoadOnly or AffineCombination needs to be chosen!"),
+                           "parameters": {"alpha": float,
+                                          "beta": float,
+                                          "gamma": float}},
+            "algorithm": {
+                "name": And(str, lambda d: d in ("InformAndTransfer", "BruteForce"),
+                            error="InformAndTransfer or BruteForce needs to be chosen!"),
+                Optional("parameters"): dict
+            },
+            "exodus": bool,
+            "x_procs": int,
+            "y_procs": int,
+            "z_procs": int,
+            "output_file_stem": str,
+            Optional("brute_force_optimization"): bool,
+            Optional("logging_level"): And(str, Use(str.lower), lambda f: f in ("info", "debug", "warning", "error"),
+                                           error="info or debug or warning or error needs to be chosen!"),
+            Optional("terminal_background"): And(str, Use(str.lower), lambda g: g in ("light", "dark"),
+                                                 error="light or dark needs to be chosen!"),
+            Optional("output_dir"): str,
+            Optional("generate_multimedia"): bool
+        })
+        self.__from_data = Schema({"data_stem": str, "phase_id": int})
+        self.__from_samplers = Schema({
+            "n_objects": int,
+            "n_mapped_ranks": int,
+            "communication_degree": int,
+            "time_sampler": {"name": And(str, Use(str.lower), lambda a: a in ("uniform", "lognormal"),
+                                         error="uniform or lognormal needs to be chosen!"),
+                             "parameters": [float, float]},
+            "volume_sampler": {"name": And(str, Use(str.lower), lambda b: b in ("uniform", "lognormal"),
+                                           error="uniform or lognormal needs to be chosen!"),
+                               "parameters": [float, float]}
+        })
+        self.__algorithm = {
+            "InformAndTransfer": Schema(
+                {"name": "InformAndTransfer",
+                 "parameters": {"n_iterations": int, "n_rounds": int, "fanout": int, "order_strategy": And(
+                     str, Use(str.lower), lambda e: e in ("arbitrary", "element_id", "increasing_times",
+                                                          "decreasing_times", "increasing_connectivity",
+                                                          "fewest_migrations", "small_objects"),
+                     error="arbitrary or element_id or increasing_times or decreasing_times or increasing_connectivity"
```

Comment:
This list of available strategies should not be hard-coded here. It should be defined somewhere else in a global list, akin for instance to the allowable keys.

Reply:
Maybe as a first approach, just define the following tuple:
```
allowed_strategies = ("arbitrary", "element_id", "increasing_times","decreasing_times", "increasing_connectivity", "fewest_migrations", "small_objects")
```
and then you can test for `e in allowed_strategy` and for the message you can do `" or ".join(allowed_strategies)`

Reply:
I will put the allowed values in `configurationValidator.py`, we could put them into other place if it exists.